checkstyle-tester/diff.groovy

```diff
@@ -628,36 +625,39 @@ def getProjectsStatistic(diffDir) {
     return projectsStatistic
 }
 
-def runMavenExecution(srcDir, excludes, checkstyleConfig,
-                      checkstyleVersion, extraMvnRegressionOptions) {
-    println "Running 'mvn clean' on $srcDir ..."
-    def mvnClean = "mvn -e --no-transfer-progress --batch-mode clean"
-    executeCmd(mvnClean)
-    println "Running Checkstyle on $srcDir ... with excludes {$excludes}"
-    def mvnSite = "mvn -e --no-transfer-progress --batch-mode site " +
-        "-Dcheckstyle.config.location=$checkstyleConfig -Dcheckstyle.excludes=$excludes"
-    if (checkstyleVersion) {
-        mvnSite = mvnSite + " -Dcheckstyle.version=$checkstyleVersion"
+def runCliExecution(allJar, mainClassOptions) {
+    def saveLoc = new File(mainClassOptions.saveDir)
+    if (!saveLoc.exists()) {
+        saveLoc.mkdirs()
     }
-    if (extraMvnRegressionOptions) {
-        mvnSite = mvnSite + " "
 
-        if (!extraMvnRegressionOptions.startsWith("-")) {
-            mvnSite = mvnSite + "-"
+    println "Running Checkstyle CLI on ${mainClassOptions.srcDir} ... with excludes {${mainClassOptions.excludes}}"
+    def cliCommand = "java -jar $allJar -c ${mainClassOptions.checkstyleConfig} " +
+        "${mainClassOptions.srcDir} -f xml -o " +
+        getOsSpecificPath("${mainClassOptions.saveDir}", "checkstyle-result.xml") +
+        " -e " + getOsSpecificPath("${mainClassOptions.srcDir}", ".git") +
+        " -e " + getOsSpecificPath("${mainClassOptions.srcDir}", ".hg")
+
+    if (mainClassOptions.excludes.length() > 0) {
```

Comment:
Why do we need length check?

Reply:
I believe this was because groovy will still create a single entry array when doing the `split` adding an `-x` to the command line with no parameter to exclude.